consumers/slack/main_test.go

```diff
@@ -0,0 +1,53 @@
+package main
```

Comment:
I'd like to see better coverage from the tests here. It'll be easier to achieve this if you split things in to their own packages.